### PR TITLE
Forward pinecone changes from azureml-rag 

### PR DESCRIPTION
**Pull request by datdo-msft**
# Description
Forward pinecone changes from azureml-rag 

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by tot0 — APPROVED** [review]

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py [hunk 1]
```diff
@@ -154,6 +156,47 @@ def build_index(
...
+                "connection_type": "environment",
+                "connection": {"key": "PINECONE_API_KEY"}
+            }
+        else:
+            pinecone_connection = get_connection_by_id_v2(index_config.pinecone_connection_id)
+            pinecone_args = {
+                **pinecone_args,
+                **{
+                    "environment": pinecone_connection["properties"]["metadata"]["environment"]
+                }
```
Comment (shpengmsft) [inline sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py:181]: import get_metadata_from_connection from azure.ai.generative.index._utils.connections

```suggestion
                    "environment": get_metadata_from_connection(pinecone_connection)["environment"]
                }
```

File: sdk/ai/azure-ai-resources/azure/ai/resources/index/_dataindex/entities/_builders/data_index_func.py [hunk 2]
```diff
@@ -244,23 +244,30 @@ def data_index_incremental_update_acs(
     from azure.ai.resources.index._dataindex.dsl._pipeline_decorator import pipeline
 
     crack_and_chunk_and_embed_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_CRACK_AND_CHUNK_AND_EMBED)
-    update_acs_index_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_UPDATE_ACS_INDEX)
+
+    if data_index.index.type == DataIndexTypes.ACS:
+        update_index_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_UPDATE_ACS_INDEX)
+    elif data_index.index.type == DataIndexTypes.PINECONE:
+        update_index_component = get_component_obj(ml_client, LLMRAGComponentUri.LLM_RAG_UPDATE_PINECONE_INDEX)
+    else:
```
Comment (shpengmsft) [inline sdk/ai/azure-ai-resources/azure/ai/resources/index/_dataindex/entities/_builders/data_index_func.py:252]: For my learning purpose, it seems we don't support FAISS index update, correct?

**Review by shpengmsft — APPROVED** [review]

File: sdk/ai/azure-ai-resources/azure/ai/resources/index/_dataindex/entities/_builders/data_index_func.py (hunk 2 above)
Comment (datdo-msft) [reply sdk/ai/azure-ai-resources/azure/ai/resources/index/_dataindex/entities/_builders/data_index_func.py:252]: We do, look for `data_index_faiss` as we handle FAISS differently. I updated this method to work for all hosted indexes (which at this point, are only ACS and Pinecone) hence why we only accept ACS and Pinecone here.

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py (hunk 1 above)
Comment (datdo-msft) [reply sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py]: updated

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py [hunk 3]
```diff
@@ -32,16 +33,17 @@ def build_index(
     embeddings_cache_path: str = None,
     ######## data source info ########
     index_input_config: Union[ACSSource, LocalSource] = None,
-    acs_config: ACSOutputConfig = None,  # todo better name?
+    index_config: Union[ACSOutputConfig, PineconeOutputConfig] = None,  # todo better name?
```
Comment (datdo-msft) [inline sdk/ai/azure-ai-generative/azure/ai/generative/index/_build_mlindex.py:36]: For reviewers, would this be considered breaking existing functionality?

File: sdk/ai/azure-ai-resources/azure/ai/resources/client/_ai_client.py [hunk 4]
```diff
@@ -257,7 +257,7 @@ def build_index_on_cloud(
         embeddings_model="text-embedding-ada-002",
         aoai_connection_id: str = DEFAULT_OPEN_AI_CONNECTION_NAME,
         ######## ACS index info ########
-        acs_config: ACSOutputConfig = None,  # todo better name?
+        index_config: Union[ACSOutputConfig, PineconeOutputConfig] = None,  # todo better name?
```
Comment (datdo-msft) [inline sdk/ai/azure-ai-resources/azure/ai/resources/client/_ai_client.py:260]: Same as in azure-ai-generative, would this be considered breaking existing functionality?

**Pull request closed without merging**